cpp/src/arrow/ipc/reader.cc

```diff
@@ -449,7 +449,7 @@ Result<std::shared_ptr<RecordBatch>> LoadRecordBatchSubset(
     const flatbuf::RecordBatch* metadata, const std::shared_ptr<Schema>& schema,
     const std::vector<bool>& inclusion_mask, const DictionaryMemo* dictionary_memo,
     const IpcReadOptions& options, MetadataVersion metadata_version,
-    Compression::type compression, io::RandomAccessFile* file) {
+    Compression::type compression, io::RandomAccessFile* file, bool swap_endian = false) {
```

Comment:
Can we put the `swap_endian` to `IpcReadOptions` from function argument?

Reply:
I'm not sure `swap_endian` is useful.
`native_endian` may be more useful.

Reply:
From a technical perspective, we can naively put `native_endian` into `IpcReadOptions` in `reader.cc`.   

From a semantics perspective, now, `native_endian` is to show the state on whether the endianness in the schema is equal to the endianness of the CPU. Since values in `IpcReadOptions` can be set by users, its flag should be like `force_naive_endian` to ask to change the endianness of the schema.   
If we put `force_naive_endian` in `IpcReadOptions`, we need to keep `native_endian` somewhere.

What do you think?

Reply:
`force_native_endian` naming is OK to me. (It should be `false` by default.)
`use_native_endian` may also be good because we already have `use_threads`.

Reply:
I see. Thus, we will have `use_native_endian` in `IpcReadOptions`. In addition, keep `swap_endian` and `swap_endian_` as now.

Reply:
Can we remove `swap_endian` and `swap_endian_` by using `IpcReadOptions::use_native_endian`?
Should we keep `swap_endian` and `swap_endian_`?

Reply:
I think we need to keep `swap_endian` and `swap_endian_`.   
IMHO, this is because `IpcReadOptions::use_native_endian` is a flag set by a user to determine whether the endianness of the schema will be changed or not if the endianness of the schema is not equal to the platform-native endianness.    
`swap_endian_` is to show we need to actually swap endianness of each element.

For example, there is a case that `IpcReadOptions::use_native_endian=true` and `swap_endian_=false` if x86 platform gets the schema with the endianness = LITTLE.

Or, can we use `IpcReadOptions::use_native_endian` as a working area o show we need to actually swap endianness of each element?

Reply:
I realized that we can drop `swap_endian` and `swap_endian_`. Then, I added `IpcReadOptions::use_native_endian` for a programmer to control whether the endian of data elements must be changed to native endian or not.

Reply:
I see.
How about adding a new simple internal struct (`struct IpcReadContext`?) for organizing options (`IpcReadOptions`, `DictionaryMemo` and `swap_endian` for now)?

I think that many arguments API isn't good to maintain.